src/components/MerchantsPage/ContributionBar.tsx

```diff
@@ -1,43 +1,87 @@
 import * as React from 'react';
-import styled from 'styled-components';
+import { useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
+import styled from 'styled-components';
 
+// We assume in this code that all amounts are > 0.
 interface Props {
-  totalDonations: number;
-  totalGiftCards: number;
+  donationsRaised: number;
+  giftAMealAmountRaised: number;
+  giftCardAmountRaised: number;
 }
 
-const ContributionBar = ({ totalDonations, totalGiftCards }: Props) => {
+const percentage = (raised: number, total: number) => raised / total * 100;
+
+const ContributionBar = ({
+  donationsRaised,
+  giftAMealAmountRaised,
+  giftCardAmountRaised,
+}: Props) => {
   const { t } = useTranslation();
 
-  const progressWidth = (raised: number, total: number) => {
-    if (raised < total) return (raised / total) * 100;
-    return 100;
-  };
+  const totalRaised = useMemo(
+    () => donationsRaised + giftAMealAmountRaised + giftCardAmountRaised,
+    [donationsRaised, giftAMealAmountRaised, giftCardAmountRaised],
+  );
+
+  const contributionBarProgress = useMemo(
+    () => ({
+      donationsRaised: percentage(donationsRaised, totalRaised),
+      giftAMealAmountRaised: percentage(giftAMealAmountRaised, totalRaised),
+      giftCardAmountRaised: percentage(giftCardAmountRaised, totalRaised),
+    }),
+    [donationsRaised, giftAMealAmountRaised, giftCardAmountRaised, totalRaised],
+  );
 
   return (
     <Container>
       <Heading>
         {t('contributionBar.header')}: $
-        {Math.floor((totalDonations + totalGiftCards) / 100).toLocaleString()}
+        {Math.floor((totalRaised) / 100).toLocaleString()}
       </Heading>
       <Contributions
         style={{
-          background: `linear-gradient(-45deg, #dd678a ${progressWidth(
-            totalDonations,
-            totalDonations + totalGiftCards
-          )}%, #F6B917 0%)`,
+          // Because we want the slash between donation types to be up/right, we need
```

Comment:
Love how well documented all this code is!

Reply:
thanks!